fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/AvailableFields.java

```diff
@@ -117,9 +117,9 @@ public static AvailableFields fromIndex(@Nonnull RecordType recordType,
         }
 
         // Like FDBRecordStoreBase.indexEntryKey(), but with key expressions instead of actual values.
-        final List<KeyExpression> primaryKeys = commonPrimaryKey == null
-                                                ? Collections.emptyList()
-                                                : commonPrimaryKey.normalizeKeyForPositions();
+        final List<KeyExpression> primaryKeys = new ArrayList<>(commonPrimaryKey == null
```

Comment:
Is there an issue tracking the need for a mutable list here?

Reply:
There is none.

Reply:
Let me know if I should create one as you only run into this problem if your `commonPrimaryKey` is null and that may be not super straightforward to get yourself into.